### PR TITLE
[TF] Enable optional run metadata output in TF runtime for debugging

### PR DESCRIPTION
**Pull request by rxwei**
[Run metadata](https://www.tensorflow.org/api_docs/python/tf/RunMetadata) in TensorFlow provides detailed execution information and is useful for debugging. As part of the profiling effort, we add support for this feature in the compiler runtime.

The runtime looks for an environment variable `SWIFT_TENSORFLOW_RUN_METADATA_OUTPUT` which specifies the file path to write the runtime metadata protobuf to. Then we can use Python to load file this as a `tf.RunMetadata` and get a timeline of execution.

This patch is based on https://github.com/tensorflow/tensorflow/commit/f943f0c2012e3820ddfe0fa86f002a80998e1faf, so the TensorFlow rev in update-checkout has been bumped up.

Also includes a few minor Swift style lints.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

**Comment by rxwei** [comment]
@swift-ci please clean test tensorflow

File: stdlib/public/TensorFlow/Utilities.swift [hunk 1]
```diff
@@ -126,6 +126,17 @@ func debugLog(_ message: @autoclosure () -> String,
   }
 }
 
+//===----------------------------------------------------------------------===//
+// File writing
+//===----------------------------------------------------------------------===//
+
+func writeContents(of buffer: UnsafePointer<TF_Buffer>,
```
Comment (mhong) [inline stdlib/public/TensorFlow/Utilities.swift]: add a doc string?

File: stdlib/public/TensorFlow/CompilerRuntime.swift [hunk 2]
```diff
@@ -536,25 +546,41 @@ extension TFState {
...
       exit(-1)
     }
     debugLog("Done running TF computation.")
+    
+    // If run metadata path was set, dump the run metadata proto to a file.
+    if let path = _RuntimeConfig.runMetadataOutputPath {
+      TF_DeleteBuffer(runOptions)
+      writeContents(of: runMetadataOutput!, toFile: path)
+      TF_DeleteBuffer(runMetadataOutput)
+      debugLog("Writing run metadata to \"\(path)\"")
```
Comment (mhong) [inline stdlib/public/TensorFlow/CompilerRuntime.swift]: nit: consider logging this before the writeContents() call.

**Review by mhong — APPROVED** [review]
Nice!

File: stdlib/public/TensorFlow/Utilities.swift (hunk 1 above)
Comment (rxwei) [reply stdlib/public/TensorFlow/Utilities.swift]: Done.

File: stdlib/public/TensorFlow/CompilerRuntime.swift (hunk 2 above)
Comment (rxwei) [reply stdlib/public/TensorFlow/CompilerRuntime.swift]: Done.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
Thanks!

**Pull request merged**